src/syngen/ml/processors/processors.py

```diff
@@ -0,0 +1,368 @@
+            *args
+    ):
+        """
+        Save the input data to the predefined path
+        """
+        DataLoader(path_to_input_data).save_data(path_to_input_data, flattened_data)
+
+    def _handle_json_columns(self):
+        """
+        Preprocess the data contained JSON columns before the training process
+        """
+        metadata = self._fetch_metadata()
+        flatten_metadata = dict()
+        for table, settings in metadata.items():
+            if table == "global":
+                continue
+            path_to_input_data = (f"{PATH_TO_MODEL_ARTIFACTS}/tmp_store/{slugify(table)}/"
+                                  f"input_data_{slugify(table)}.pkl")
+            source = settings.get("train_settings", {}).get("source", "")
+            data, schema = self._load_source(source, table)
+            order_of_columns = data.columns.to_list()
+            self._remove_existed_artifacts(table_name=table)
+            self._prepare_dirs(table)
+            if json_columns := self._get_json_columns(data):
+                logger.info(
+                    f"The table '{table}' contains JSON columns: {', '.join(json_columns)}"
+                )
+                logger.info(f"Flattening the JSON columns in the table - '{table}'")
+                (flattened_data,
+                 flattening_mapping,
+                 duplicated_columns) = self._get_flattened_df(data, json_columns)
+                self._save_input_data(path_to_input_data, flattened_data, table)
+                flatten_metadata[table] = {
+                    "flattening_mapping": flattening_mapping,
+                    "duplicated_columns": duplicated_columns,
+                    "order_of_columns": order_of_columns
+                }
+                logger.info(f"The table '{table}' has been successfully flattened")
+        if flatten_metadata:
+            self._save_flatten_metadata(flatten_metadata)
+
+    def run(self):
+        """
+        Launch the preprocessing process:
+        """
+        self._run_script()
+        self._handle_json_columns()
+
+
+class PostprocessHandler(Processor):
+    """
+    Class for the postprocessing of the generated data
+    """
+
+    def __init__(
+            self,
+            metadata_path: Optional[str],
+            table_name: Optional[str],
+            settings: Dict
+    ):
+        super().__init__(metadata_path, table_name, settings)
+
+    def _fetch_flatten_config(self, table_name: str) -> Dict:
+        """
+        Fetch the metadata of the flattening process
+        """
+        with open(self.path_to_flatten_metadata, "r") as f:
+            flatten_metadata = json.load(f).get(table_name)
+            return flatten_metadata
+
+    @staticmethod
+    def _check_none_values(x) -> bool:
+        """
+        Check if the value is None or np.NaN
+        """
+        return x is None or (isinstance(x, float) and np.isnan(x))
+
+    def _remove_none_from_struct(self, input_data: Any):
```

Comment:
Why not staticmethod?

Reply:
It's a staticmethod because it doesn't use the instance of the class